script.rb

```diff
@@ -0,0 +1,155 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    counter = 0
+    while counter < size
+      yield(self[counter])
+      counter += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    counter = 0
+    while counter < size
+      yield(self[counter], counter)
+      counter += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    arr = []
+    my_each do |x|
+      arr << x if yield(x)
+    end
+    arr
+  end
+
+  def my_all?(args = nil)
+    return true unless block_given?
+
+    my_each do |x|
+      return false unless yield(x)
+    end
+    if args.is_a?(Class)
+      my_each do |x|
+        return false unless x.is_a?(args)
+      end
+    elsif pattern.is_a?(Regexp)
+      my_each do |x|
+        return false unless args.match(x.to_s)
+      end
+    else
+      my_each do |x|
+        return false unless x == args
+      end
+    end
+  end
+
+  def my_any?(args = nil)
+    return false unless block_given?
+
+    my_each do |x|
+      return true if yield(x)
+    end
+    if args.is_a?(Class)
+      my_each do |x|
+        return true if x.is_a?(args)
+      end
+    elsif args.is_a?(Regexp)
+      my_each do |x|
+        return true if args.match(x.to_s)
+      end
+    else
+      my_each do |x|
+        return true if x == args
+      end
+    end
+  end
```

Comment:
### Your `my_any` method are failing in below tests
- [ ] enumerables my_any? returns true if the block ever returns a value other than false or nil
- [ ]  enumerables my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
- [ ] enumerables my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class
- [ ] enumerables my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
- [ ] enumerables my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern